distributed/scheduler.py

```diff
@@ -4505,4 +4512,7 @@ def heartbeat_interval(n):
 
 
 class KilledWorker(Exception):
-    pass
+    def __init__(self, *args, **kwargs):
+        super(KilledWorker, self).__init__(*args)
+        self.args = args
+        self.kwargs = kwargs
```

Comment:
If we do as above and include just a single keyword `last_failed_worker` or something then it would be good to include that explicitly, rather than catch everything.